azure_monitor/src/azure_monitor/sdk/auto_collection/request_metrics.py

```diff
@@ -1,19 +1,74 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 import logging
+import threading
 import time
+from http.server import HTTPServer
 from typing import Dict
 
 from opentelemetry.metrics import Meter, Observer
 from opentelemetry.sdk.metrics import UpDownSumObserver
 
-from azure_monitor.sdk.auto_collection.metrics_span_processor import (
-    AzureMetricsSpanProcessor,
-)
 from azure_monitor.sdk.auto_collection.utils import AutoCollectionType
 
+_requests_lock = threading.Lock()
 logger = logging.getLogger(__name__)
 requests_map = dict()
+ORIGINAL_CONSTRUCTOR = HTTPServer.__init__
+
+
+def request_patch(func):
+    def wrapper(self=None):
+        start_time = time.time()
+        func(self)
+        end_time = time.time()
+
+        with _requests_lock:
+            # Update Count
+            count = requests_map.get("count", 0)
+            requests_map["count"] = count + 1
+            # Update duration
+            duration = requests_map.get("duration", 0)
+            requests_map["duration"] = duration + (end_time - start_time)
+
+    return wrapper
+
+
+def send_response_patch(func):
+    def wrapper(self, code, message=None):
+        func(self, code, message)
+        if code >= 200 < 300:
+            with _requests_lock:
```

Comment:
With a single lock holding requests sent and responses received, responses from different requests could possible be waiting on different requests that are being sent. We need a way to have the returned result not depend on the calculations of telemetry.